x-pack/plugins/security/public/nav_control/nav_control_component.tsx

```diff
@@ -66,6 +69,27 @@ export class SecurityNavControl extends Component<Props, State> {
   componentDidMount() {
     this.subscription = this.props.userMenuLinks$.subscribe(async (userMenuLinks) => {
       this.setState({ userMenuLinks });
+
+      if (userMenuLinks.length) {
```

Comment:
__question:__ do we really need to alter state here? Can't we just check if we have a link with `setAsProfile` right before we render default profile link in `render`? The perf impact seems to be negligible. Something like this: 
```ts
if (!isAnonymousUser) {
  const hasCustomProfileLinks = userMenuLinks.some(({ setAsProfile }) => setAsProfile === true);
  const profileMenuItem = {
    name: (
      <FormattedMessage
        id="xpack.security.navControlComponent.editProfileLinkText"
        defaultMessage="{profileOverridden, select, true{Preferences} other{Profile}}"
        values={{ profileOverridden: hasCustomProfileLinks }}
      />
    ),
.....
```

--------------

Regarding multiple links with `setAsProfile`, I believe we just should throw error (optionally with detailed explanation that link with `labelA` is already used as a profile link) in `addUserMenuLinks` in `nav_control_service.tsx` right at the moment when something wants to add a link with `setAsProfile` when we already have one (the service tracks all links so it should be easy to figure out). People tend to ignore console warnings anyway :slightly_smiling_face: 

--------------

Ideally I'd rather expose dedicated methods to manage (hide/rename) built-in links like Profile and Logout, but I agree that it doesn't make sense to generalize this right now since everything is still in flux, so :+1: to the `setAsProfile` approach.

Reply:
I've removed the state stuff and am now throwing errors for two situations: 1) a custom profile link is already set (the name and href for the existing custom profile link are displayed) and 2) there is no custom profile link and the plugin is passing more than one (the total number passed is displayed).